### DISCUSSION TITLE
Add a "last chance" flag and page

### DISCUSSION THREAD

**Discussion by marionf** [PrestaShop/PrestaShop#40165] [Ideas]
#### Is your feature request related to a problem?

As a merchant, I want to highlight products that will be soon out of stock (but which do not necessarily have a reduced price) and that will not be back into the stock after.

#### Describe the solution you'd like

1) Create a "Last chance!" flag
2) Create a "last-chance" page
3) Add a checkbox to flag product as a "Last chance!" and add them on the "last-chance" page

#### Specs
"last-chance" page could be managed in the BO > Shop parameters > Traffic & SEO > SEO & URLs
The default title of the page will be "Last chance". The default friendly URL of the page will be "last-chance"
The default meta title of the page will be: 'Last chance'
The default meta description of the page will be: 'Soon, the products on this page will be gone. This is your last chance to buy them, don't miss it!' in Shop.Theme.Catalog

For **standard and packs products**, add on the BO product page, in the basic settings tab and in the quantities tab, under the "quantity" field a checkbox: 
Label: 'Display the 'Last chance' flag' in Admin.Catalog.Feature
Tooltip: 'Use this flag to highlight products that will not be restocked. The product will appear on the 'Last chance' page.' in Admin.Catalog.Help
For **virtual products**, add on the BO product page, in the basic settings tab and in the virtual product tab, under the "quantity" field a checkbox: 
Label: 'Display the 'Last chance' flag' in Admin.Catalog.Feature
Tooltip: 'Use this flag to highlight products that will not be restocked. The product will appear on the 'Last chance' page.' in Admin.Catalog.Help
For **products with combinations**, add on the BO product page, in the combinations tab, under the "quantity" field a checkbox:
Label: 'Display the 'Last chance' flag' in Admin.Catalog.Feature
Tooltip: 'Use this flag to highlight products that will not be restocked. The product will appear on the 'Last chance' page.' in Admin.Catalog.Help

When the checkbox 'Display the 'Last chance' flag'  is checked, the product will be displayed on the "last-chance" page and the "Last chance!" flag will be displayed on the product page and product listings.

**For products with combinations :** 
- The flag should be displayed according to the selected combination
- All the combinations will be displayed on the "last-chance" page

**Example :**
Tshirt - S - default combination -  last chance box not checked 
Tshirt - M - last chance box is checked 
Thsirt - L - last chance box is checked 

Product list: "Last chance!" flag not displayed because the default combination S isn't checked as last chance
Product page: "Last chance!" flag displayed when the combination M and L are selected
Last chances page: The products Tshirt - M and Thsirt - L are displayed with the "Last chance!" flag 

#### Alternatives you've considered

This issue should be done at the same time or just after this one: https://github.com/PrestaShop/PrestaShop/issues/24153

**Comment by Hlavtox** [comment]
Hi @marionf, I suggest to:

- Rename the original **On sale** (you renamed to **Price drop**) to **Last chance**, it will serve the original purpose and have better wording. Or just revert the rename to not make people confused. They can rename the flag how they want, can't they?
- Just forget the **Price drop** flag. as @Julievrz said `at least in French since 'On sale' means 'Promotion': that would be weird to have a promotion flag if on a page with products that have no reduction => and that's illegal in France anyway.` - what is the purpose of the "Price drop" flag now? Isn't it "illegal" if it can be checked even with no reduction? :-) If product is discounted, isn't it clearly displayed by the red flag with discount amount or discount percentage?

If you keep it named "Price drop" it will confuse a lot of users. They will check it and the product will not show up on Price drop page.

**Comment by marionf** [comment]
Hi @Hlavtox 

> If you keep it named "Price drop" it will confuse a lot of users.

I made a [new PR](https://github.com/PrestaShop/PrestaShop/pull/24151), we are going to keep "On sale" flag for now 

> what is the purpose of the "Price drop" flag now?

I opened [this issue](https://github.com/PrestaShop/PrestaShop/issues/24153) for the "On sale" flag. The purpose is to link it with the specific prices as "on sale" means that the product has a reduced price.

>  isn't it clearly displayed by the red flag with the discount amount or discount percentage?

Some merchants might want to put forward some specifics products with the reduced price.

> Just forget the Price drop flag. Rename the original On sale (you renamed to Price drop) to Last chance, it will serve the original purpose and have better wording.

This works for your use case, what if a merchant has some products discounted and some products at the end of stock but without discount? He/she might want to put them forward by displaying the "On sale !" flag on all or just some discounted products and the "last chance !" flag on the end of stock products.
